src/app/exam/Examinees.js

```diff
@@ -6,62 +6,81 @@ import {examService} from "../../services/services";
 import {ItemListPage} from "../commons/ItemListPage/ItemListPage";
 import {DropDownBtn} from "../commons/buttons/DropDownButton";
 import {AiOutlineMail, AiOutlineUsergroupAdd} from "react-icons/ai";
-import {AddExamineeModal} from "./modals/AddExamineeModal";
+import {AddExamineesModal} from "./modals/AddExamineesModal";
 import {ThreeDotsButton} from "../commons/buttons/ThreeDotsButton";
 import {RemoveConfirmationModal} from "../commons/modals/RemoveConfirmationModal";
 import {Spinner} from "../commons/Spinner";
 
 const Examinees = withRouter(({history, match}) => {
         const currentPathName = history.location.pathname;
-        const [students, setStudents] = useState(undefined);
-        const [examinee, setExaminee] = useState(undefined);
+        const [examinees, setExaminees] = useState(undefined);
+        const [selectedExaminee, setSelectedExaminee] = useState(undefined);
         const [exam, setExam] = useState(undefined);
         const [showAddStudentModal, setShowAddStudentModal] = useState(false);
         const [showAddGroupModal, setShowAddGroupModal] = useState(false);
         const [showRemoveExamineeConfirmationModal, setShowRemoveExamineeConfirmationModal] = useState(false);
 
         const examId = match.params.examId;
 
+        useEffect(() => {
+            if (!exam) {
+                examService.getExam(examId)
+                    .then(exam => setExam(exam))
+            }
+            if (!examinees && exam) {
+                console.log("in effect")
+                examService.getExaminees(examId)
+                    .then(students => setExaminees(students));
+            }
+        }, [exam, examId, examinees]);
+
         const actionItemsButton = ({examinee}) => new ThreeDotsButton({
-            dropDownItems: [
-                {
-                    name: "Remove",
-                    dangerous: true,
-                    onClick: () => {
-                        setShowRemoveExamineeConfirmationModal(true)
-                        setExaminee(examinee)
-                    }
+            dropDownItems: [{
+                name: "Remove",
+                dangerous: true,
+                onClick: () => {
+                    setShowRemoveExamineeConfirmationModal(true)
+                    setSelectedExaminee(examinee)
                 }
-            ]
+            }]
         })
 
-        const addExaminees = (emails) => {
-            examService.addExaminees(exam.id, emails)
-                .then(errorList => console.log(`errorList=${errorList}`))
+        const addExaminees = async (emails) => {
+            await examService.addExaminees(exam.id, emails)
+                .then(errorList => {
+                    console.log(`errorList=${errorList}`);
+                })
+            examService.getExaminees(examId)
+                .then(students => {
+                    // TODO: remove setting empty array and fixed "render more hooks than expected problems"
```

Comment:
Not clear enough
`// TODO: currently, to avoid "render fewer hooks than expected" error thrown from React, setting an empty array is a effective trick, but we need to know the root cause and use the more proper way instead.`